test/e2e-tests.sh

```diff
@@ -55,40 +58,53 @@ function knative_setup() {
   echo "Installing Knative Eventing"
   ko apply -f ${EVENTING_CONFIG} || return 1
   wait_until_pods_running knative-eventing || fail_test "Knative Eventing did not come up"
+}
+
+# Teardown the Knative environment after tests finish.
+function knative_teardown() {
+  echo ">> Stopping Knative Eventing"
+  echo "Uninstalling Knative Eventing"
+  ko delete --ignore-not-found=true --now --timeout 60s -f ${EVENTING_CONFIG}
+}
 
+# Setup resources common to all eventing tests.
+function test_setup() {
+  # Install provisioners used by the tests.
   echo "Installing In-Memory ClusterChannelProvisioner"
   ko apply -f ${IN_MEMORY_CHANNEL_CONFIG} || return 1
   wait_until_pods_running knative-eventing || fail_test "Failed to install the In-Memory ClusterChannelProvisioner"
 
-  E2E_PROJECT_ID="$(gcloud config get-value project)"
   echo "Installing GCPPubSub ClusterChannelProvisioner"
-  gcppubsub_setup
+  gcppubsub_setup || return 1
   sed "s/REPLACE_WITH_GCP_PROJECT/${E2E_PROJECT_ID}/" ${GCP_PUBSUB_CONFIG_TEMPLATE} > ${GCP_PUBSUB_CONFIG}
-  ko apply -f ${GCP_PUBSUB_CONFIG}
+  ko apply -f ${GCP_PUBSUB_CONFIG} || return 1
   wait_until_pods_running knative-eventing || fail_test "Failed to install the GCPPubSub ClusterChannelProvisioner"
-}
 
-# Teardown the Knative environment after tests finish.
-function knative_teardown() {
-  echo ">> Stopping Knative Eventing"
-  echo "Uninstalling Knative Eventing"
-  ko delete --ignore-not-found=true --now --timeout 60s -f ${EVENTING_CONFIG}
+  echo "Installing NATSS ClusterChannelProvisioner"
+  natss_setup || return 1
```

Comment:
Only the last `kubectl apply` in `natss_setup` will return 1 in case of error. I suggest you fail directly in the function instead of here. Here and everywhere.